plugin/procfs/procfs.cc

```diff
@@ -0,0 +1,475 @@
+{
+  if (cond != 0 && cond->type() == Item::FUNC_ITEM)
+  {
+    Item_func *func= static_cast<Item_func *>(cond);
+    if (func != NULL && func->functype() == Item_func::LIKE_FUNC)
+    {
+      Item_func_like *like_func= static_cast<Item_func_like *>(func);
+      if (like_func->arguments()[0]->type() == Item::FIELD_ITEM &&
+          my_strcasecmp(system_charset_info,
+                        like_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0)
+      {
+        char    buff[1024];
+        String *res;
+        String  filter(buff, sizeof(buff), system_charset_info);
+        if (like_func->arguments()[1] != NULL &&
+            (res= like_func->arguments()[1]->val_str(&filter)))
+        {
+          like_arg->append(res->c_ptr_safe(), res->length());
+          return false;
+        }
+      }
+    }
+  }
+  return true;
+}
+
+
+static bool get_in_condition_argument(Item *                       cond,
+                                      std::map<std::string, bool> &in_args,
+                                      const std::string &          field_name)
+{
+  if (cond != 0 && cond->type() == Item::FUNC_ITEM)
+  {
+    Item_func *func= static_cast<Item_func *>(cond);
+    if (func != NULL && func->functype() == Item_func::IN_FUNC)
+    {
+      Item_func_in *in_func= static_cast<Item_func_in *>(func);
+      if (in_func->arguments()[0]->type() == Item::FIELD_ITEM &&
+          my_strcasecmp(system_charset_info,
+                        in_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0)
+      {
+        char    buff[1024];
+        String *res;
+        String  filter(buff, sizeof(buff), system_charset_info);
+        for (uint i= 1; i < in_func->arg_count; ++i)
+        {
+          if (in_func->arguments()[i] != NULL &&
+              (res= in_func->arguments()[i]->val_str(&filter)) &&
+              res->length() > 0)
+          {
+            in_args[std::string(res->c_ptr_safe(), res->length())]= true;
+          }
+        }
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void fill_procfs_view_row(THD *thd, TABLE *table, const char *fname,
+                                 char *buf, size_t sz)
+{
+  if (sz == 0)
+    return;
+
+  table->field[0]->store(fname, strlen(fname), system_charset_info);
+  table->field[1]->store(buf, sz, system_charset_info);
+  schema_table_store_record(thd, table);
+}
+
+static std::vector<std::string> limited_glob_files(std::string path,
+                                                   std::string pattern,
+                                                   int         max_results)
+{
+  std::vector<std::string> files_found;
+
+  if (max_results <= 0)
+    return files_found;
+
+  DIR *          dir= opendir(path.c_str());
+  struct dirent *dir_entry;
+
+  if (!dir)
+    return files_found;
+
+  std::vector<char> real_file_path(PATH_MAX);
```

Comment:
No need to allocate this dynamically. char real_file_path[PATH_MAX] is fine, or std::array if switching to C++11.